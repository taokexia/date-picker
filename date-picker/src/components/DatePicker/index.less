@import './vars.less';

.@{prefix-cls} {
  position: relative;
  display: inline-block;
  user-select: none;
  &-input {
    user-select: none;
    box-sizing: border-box;
    margin: 0;
    position: relative;
    display: inline-block;
    width: 100%;
    padding: @input-padding;
    height: @input-height;
    font-size: @font-size;
    line-height: 1.5;
    color: @font-color;
    background-color: #fff;
    background-image: none;
    border: 1px solid @border-color;
    border-radius: @border-radius;
    transition: all @default-transition;
    outline: none;

    &:active,
    &:hover,
    &:focus {
      border-color: @primary-color;
      outline: none;
      box-shadow: @input-box-shadow;
    }

    &-disabled {
      cursor: not-allowed;
      color: @disabled-font-color;
      background-color: @disabled-color;
      &:active,
      &:hover,
      &:focus {
        border-color: @border-color;
        outline: none;
        box-shadow: none;
      }
    }

    &-wrapper {
      position: relative;
      display: inline-block;
      width: @date-picker-content-width;
      user-select: none;
    }
  }
  &-inner {
    display: inline-block;
    position: relative;
    cursor: pointer;
  }
  &-panel {
    width: @date-picker-content-width;
    height: auto;
    overflow-y: auto;
    position: absolute;
    left: 0;
    top: 0;
    margin-top: @date-picker-offset-top;
    background: #fff;
    border: 1px solid @border-color;
    z-index: @date-picker-z-index;
    box-shadow: @default-shadow;
    padding: @date-picker-padding;
    border-radius: @border-radius;
    &-header {
      margin-left: 12px;
      margin-bottom: 28px;
      color: @primary-color;
      font-weight: bold;
      &-selected-date {
        font-size: @font-size + 4;
        letter-spacing: 2px;
        margin-right: 10px;
        font-weight: bold;
        user-select: none;
      }
      &-switch {
        float: right;
        &-group {
          cursor: pointer;
          margin: 0 5px;
          border: 1px solid @border-color;
          border-radius: 50%;
          background-color: @bg-color;
          text-align: center;
          width: 25px;
          height: 25px;
          line-height: 25px;
          display: inline-block;
          transition: all @default-transition;

          &:hover {
            background-color: @primary-color;
            border-color: @primary-color;
            color: #fff;
            text-shadow: @default-text-shadow;
          }
        }
      }

      &-date {
        user-select: none;
        text-align: center;
        display: inline-block;
        width: @date-picker-day-width;
        height: @date-picker-day-height;
        line-height: @date-picker-day-height;
        border-radius: 50%;
        margin: @date-picker-day-offset;
        cursor: pointer;
        border: 1px solid transparent;
        transition: all @default-transition;
        &:not(.@{prefix-cls}-day-title):not(.@{prefix-cls}-last-month):not(.@{prefix-cls}-next-month):not(.@{prefix-cls}-disabled-date):hover {
          border: 1px solid @primary-color;
        }
      }
    }
    &-content {
      display: flex;
      align-items: center;
      align-content: center;
      flex-wrap: wrap;
      .date-picker-date-item() {
        user-select: none;
        text-align: center;
        display: inline-block;
        width: @date-picker-day-width;
        height: @date-picker-day-height;
        line-height: @date-picker-day-height;
        border-radius: 50% !important;
        margin: @date-picker-day-offset;
        cursor: pointer;
        border: 1px solid transparent;
        transition: all @default-transition;
      }
      &-header {
        &-day {
          .date-picker-date-item();
          color: @font-color;
          opacity: 0.65;
          cursor: text !important;
        }
      }
      &-date {
        .date-picker-date-item();
      }
      &-prev-month,
      &-next-month {
        color: @disabled-font-color;
      }
    }
  }
  &-button {
    width: 22px;
    height: 22px;
    line-height: 22px;
    border-radius: 50%;
    background-color: @disabled-color;
    display: inline-block;
    text-align: center;
    cursor: pointer;
    &-icon {
      color: @primary-color;
    }
  }
  &-current-month {
    border: 1px solid transparent;
    border-radius: @border-radius;
  }
  &-selected-date {
    border-color: @primary-color;
    background-color: @primary-color;
    color: #fff;
    text-shadow: @default-text-shadow;
  }
}
