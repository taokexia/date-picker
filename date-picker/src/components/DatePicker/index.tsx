import React from 'react'
import DatePickerPanel from './panel'
import cls from 'classnames'
import { Moment } from 'moment'
import './index.less'

export interface DatePickerProps {
  prefixCls?: string
  getPopupContainer?: () => HTMLElement
  onChange?: (value?: Moment | undefined) => void
  showToday?: boolean
  allowClear?: boolean
  className?: string
  popupContainerClassName?: string
  style?: React.CSSProperties
  placeholder?: string
  disabled?: boolean
  value?: Moment
  defaultValue?: Moment
}

export default function DatePicker(props: DatePickerProps) {
  const { placeholder, disabled, defaultValue, value, ...attr } = props
  const { prefixCls } = props
  return (
    <>
      <div className={`${prefixCls}-input-wrapper`}>
        <input
          type="text"
          placeholder={placeholder}
          readOnly
          disabled={disabled}
          className={cls(`${prefixCls}-input`, {
            [`${prefixCls}-input-disabled`]: disabled
          })}
        />
      </div>
      <DatePickerPanel {...attr} />
    </>
  )
}

DatePicker.defaultProps = {
 prefixCls: 'date-picker',
 getPopupContainer: () => document.body,
 showToday: true,
 allowClear: true,
 disabled: false
} as DatePickerProps