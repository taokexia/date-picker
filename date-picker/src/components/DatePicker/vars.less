@prefix-cls : date-picker;
@primary-color: #31c27c;
@border-color: #e8e8e8;
@font-color: rgba(0, 0, 0, 0.65);
@border-radius: 4px;
@font-size: 14px;
@bg-color: #fafafa;
@default-shadow: 0 4px 22px 0 rgba(15, 35, 95, 0.12);
@default-section-shadow: 0 1px 4px 0 rgba(15, 35, 95, 0.12);
@default-text-shadow: 0 1px 0 rgba(0, 0, 0, 0.1);
@disabled-color: #f5f5f5;
@disabled-font-color: fade(@font-color, 25%);

@date-picker-z-index: 88;
@date-picker-content-height: 470px;
@date-picker-content-width: 322px;
@date-picker-content-max-width: 400px;
@date-picker-day-width: 36px;
@date-picker-day-height: 36px;
@date-picker-day-offset: 0 4px 4px 0;
@date-picker-padding: 15px 20px;
@date-picker-offset-top: 5px;

//Input 输入框
@input-height: 40px;
@input-height-large: 50px;
@input-height-small: 30px;
@input-padding: 4px 10px;
@input-box-shadow: 0 0 0 2px fade(@primary-color, 20%);
@input-prefix-offset: 12px;
@input-suffix-offset: 12px;

@default-transition: 500ms cubic-bezier(0.165, 0.84, 0.44, 1);

@keyframes date-picker-open {
  from {
    opacity: 0;
    transform: translate3d(0, -30px, 0);
  }
  to {
    opacity: 1;
    transform: scale(1) translate3d(0, 0, 0);
  }
}

@keyframes date-picker-close {
  from {
    opacity: 1;
    transform: translate3d(0, 0, 0);
  }
  to {
    opacity: 0;
    transform: translate3d(0, -15px, 0);
  }
}