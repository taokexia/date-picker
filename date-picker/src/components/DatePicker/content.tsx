// DatePicker/content.tsx
import React, { useMemo, useCallback } from 'react'
import moment from 'moment'
import cls from 'classnames'
import { DatePickerPanelCommonProps } from './panel'
import { DatePickerProps } from './'
import { range } from '../../utils'

interface DatePickerPanelContentProps extends DatePickerPanelCommonProps {
  selectedDate: moment.Moment
  onSelectedDate?: DatePickerProps['onChange']
}

type SelectedMonthType = 'prev' | 'current' | 'next'
const CALENDAR_HEADERS = ['一', '二', '三', '四', '五', '六', '日']
const WEEKDAY = 7
const COLUMN = 6

export function DatePickerPanelContent(props: DatePickerPanelContentProps) {
  const { prefixCls, selectedDate: selectedDateFromProps } = props
  const panelContentPrefixCls = useMemo(() => `${prefixCls}-panel-content`, [
    prefixCls
  ])

  // 当前选中的月份
  const cloneSelectedDate = useMemo(() => selectedDateFromProps.clone(), [
    selectedDateFromProps
  ])
  // 当前选中的哪一天 （用于高亮显示）
  const selectedDate = useMemo(() => cloneSelectedDate.date(), [
    cloneSelectedDate
  ])
  // 选中的这个月有多少天
  const selectedMonthVisibleDays = useMemo(
    () => cloneSelectedDate.daysInMonth(),
    [cloneSelectedDate]
  )
  // 选中的这个月第一天 （用于这一天放在星期几的位置去渲染）
  const selectedDateFirst = useMemo(() => cloneSelectedDate.date(1), [
    cloneSelectedDate
  ])
  // 上个月显示的天数
  const prevMonthVisibleDays = useMemo(
    () => selectedDateFirst.isoWeekday() - 1,
    [selectedDateFirst]
  )
  // 上一个月有多少天
  const prevMonthDays = useMemo(
    () =>
      cloneSelectedDate
        .clone()
        .subtract(1, 'month')
        .daysInMonth(),
    [cloneSelectedDate]
  )

  // 下个月显示的天数
  const nextMonthVisibleDays = useMemo(
    () => COLUMN * WEEKDAY - selectedMonthVisibleDays - prevMonthVisibleDays,
    [selectedMonthVisibleDays, prevMonthVisibleDays]
  )

  const onSelectedDate = useCallback(
    (originSelectedDate: number) => (type: SelectedMonthType) => () => {
      let selectedDate = cloneSelectedDate.date(originSelectedDate)
      switch (type) {
        case 'prev':
          selectedDate = selectedDate.subtract(1, 'month')
          break
        case 'next':
          selectedDate = selectedDate.add(1, 'month')
          break
      }
      props.onSelectedDate?.(selectedDate)
    },
    [cloneSelectedDate]
  )

  return (
    <div className={cls(panelContentPrefixCls)}>
      <div className={cls(`${panelContentPrefixCls}-header`)}>
        {CALENDAR_HEADERS.map((day) => (
          <span
            className={cls(
              `${panelContentPrefixCls}-header-day`,
              `${prefixCls}-day-title`
            )}
            key={day}
          >
            {day}
          </span>
        ))}
      </div>
      <div>
        {range(prevMonthVisibleDays).map((_, index) => {
          const currentDate = prevMonthDays - prevMonthVisibleDays + index + 1
          return (
            <span
              className={cls(
                `${panelContentPrefixCls}-date`,
                `${panelContentPrefixCls}-prev-month`
              )}
              key={`prev-month-date-${index}`}
              onClick={onSelectedDate(currentDate)('prev')}
            >
              {currentDate}
            </span>
          )
        })}
        {range(selectedMonthVisibleDays).map((_, index) => {
          const currentDate = index + 1
          return (
            <span
              className={cls(
                `${panelContentPrefixCls}-date`,
                `${prefixCls}-current-month`,
                {
                  [`${prefixCls}-selected-date`]: selectedDate === currentDate
                }
              )}
              key={`current-month-date-${index}`}
              onClick={onSelectedDate(currentDate)('current')}
            >
              {currentDate}
            </span>
          )
        })}
        {range(nextMonthVisibleDays).map((_, index) => {
          const currentDate = index + 1
          return (
            <span
              className={cls(
                `${panelContentPrefixCls}-date`,
                `${panelContentPrefixCls}-next-month`
              )}
              key={`next-month-date-${index}`}
              onClick={onSelectedDate(currentDate)('next')}
            >
              {currentDate}
            </span>
          )
        })}
      </div>
    </div>
  )
}