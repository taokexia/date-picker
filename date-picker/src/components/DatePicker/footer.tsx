import React, { useMemo } from 'react'
import cls from 'classnames'
import { DatePickerPanelCommonProps, DatePickerPanelProps } from './panel'

export interface DatePickerPanelFooterProps
  extends DatePickerPanelCommonProps,
    Pick<DatePickerPanelProps, 'allowClear' | 'showToday'> {
  onSelectToday: () => void
  onClearSelectedDate: () => void
}

export function DatePickerPanelFooter(props: DatePickerPanelFooterProps) {
  const {
    showToday,
    allowClear,
    prefixCls,
    onSelectToday,
    onClearSelectedDate
  } = props

  const panelFooterPrefixCls = useMemo(() => `${prefixCls}-panel-footer`, [
    prefixCls
  ])

  return showToday || allowClear ? (
    <div className={cls(panelFooterPrefixCls)}>
      {showToday && (
        <div
          className={cls(`${panelFooterPrefixCls}-today`)}
          onClick={onSelectToday}
        >
          今天
        </div>
      )}
      {allowClear && (
        <div
          className={cls(`${panelFooterPrefixCls}-clear`)}
          onClick={onClearSelectedDate}
        >
          清除
        </div>
      )}
    </div>
  ) : null
}