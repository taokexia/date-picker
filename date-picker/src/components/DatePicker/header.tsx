import React, { useMemo } from 'react'
import { DatePickerPanelCommonProps } from './panel'
import cls from 'classnames'
import moment from 'moment'
import { ArrowLeftIcon, ArrowRightIcon } from '../Icon'

interface DatePickerPanelHeaderProps extends DatePickerPanelCommonProps {
  selectedDate: moment.Moment
  onSubtractMonth: () => void
  onAddMonth: () => void
}

export function DatePickerPanelHeader(props: DatePickerPanelHeaderProps) {
  const { prefixCls, selectedDate, onSubtractMonth, onAddMonth } = props
  const panelHeaderPrefixCls = useMemo(() => `${prefixCls}-panel-header`, [
    prefixCls
  ])
  const selectedYear = useMemo(() => selectedDate.year(), [selectedDate])
  const selectedMonth = useMemo(() => selectedDate.month() + 1, [selectedDate])

  return (
    <div className={cls(panelHeaderPrefixCls)}>
      <span className={cls(`${panelHeaderPrefixCls}-selected-date`)}>
        <span className={cls(`${panelHeaderPrefixCls}-selected-date-year`)}>
          {selectedYear}年
        </span>
        <span className={cls(`${panelHeaderPrefixCls}-selected-date-month`)}>
          {selectedMonth}月
        </span>
      </span>
      <span className={cls(`${panelHeaderPrefixCls}-switch`)}>
        <span
          className={cls(`${panelHeaderPrefixCls}-switch-group`)}
          onClick={onSubtractMonth}
        >
          <ArrowLeftIcon />
        </span>
        <span
          className={cls(`${panelHeaderPrefixCls}-switch-group`)}
          onClick={onAddMonth}
        >
          <ArrowRightIcon />
        </span>
      </span>
    </div>
  )
}