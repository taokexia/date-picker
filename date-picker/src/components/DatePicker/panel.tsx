import React from 'react'
import { DatePickerPanelHeader } from './header'
import { DatePickerPanelFooter } from './footer'
import { DatePickerPanelContent } from './content'
import { createPortal } from 'react-dom'

function DatePickerPanel() {
  const getPopupContainer = () => document.body
  return createPortal(
    <div className={`date-picker-panel`}>
      <DatePickerPanelHeader />
      <DatePickerPanelContent />
      <DatePickerPanelFooter />
    </div>,
    getPopupContainer()
  )
}

export default DatePickerPanel