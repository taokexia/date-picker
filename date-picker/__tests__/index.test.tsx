// __tests__/index.test.tsx

import React from 'react'
import { render, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import DatePicker from '../src/components/DatePicker'
import DatePickerPanel from '../src/components/DatePicker/panel'

describe('测试 DatePicker 组件', () => {
  it('DatePicker 组件 的快照需要保持一致', () => {
    const wrapper = render(<DatePicker />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('调用 DatePicker, 渲染 input 框 和 panel 浮层', () => {
    const wrapper = shallow(<DatePicker/>)
    expect(wrapper.find('input')).toHaveLength(1)
    expect(wrapper.find(DatePickerPanel)).toHaveLength(1)
  })

  it('动用 DatePicker, onChange 不应该被执行', () => {
    const onChange = jest.fn()
    shallow(<DatePicker onChange={onChange} />)
    expect(onChange).not.toHaveBeenCalled()
  })
})