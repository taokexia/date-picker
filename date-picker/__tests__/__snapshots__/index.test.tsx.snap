// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 DatePicker 组件 DatePicker 组件 的快照需要保持一致 1`] = `
Array [
  <div
    class="date-picker-input-wrapper"
  >
    <input
      class="date-picker-input"
      readonly=""
      type="text"
      value=""
    />
    <svg
      class="date-picker-input-suffix"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        ry="2"
        width="18"
        x="3"
        y="4"
      />
      <line
        x1="16"
        x2="16"
        y1="2"
        y2="6"
      />
      <line
        x1="8"
        x2="8"
        y1="2"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="10"
        y2="10"
      />
    </svg>
  </div>,
  <div
    class="date-picker-panel date-picker-close date-picker-no-animate"
    style="left:0;top:0"
  >
    <div
      class="date-picker-panel-header"
    >
      <span
        class="date-picker-panel-header-selected-date"
      >
        <span
          class="date-picker-panel-header-selected-date-year"
        >
          2019年
        </span>
        <span
          class="date-picker-panel-header-selected-date-month"
        >
          12月
        </span>
      </span>
      <span
        class="date-picker-panel-header-switch"
      >
        <span
          class="date-picker-panel-header-switch-group"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="15 18 9 12 15 6"
            />
          </svg>
        </span>
        <span
          class="date-picker-panel-header-switch-group"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="9 18 15 12 9 6"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      class="date-picker-panel-content"
    >
      <div
        class="date-picker-panel-content-header"
      >
        <span
          class="date-picker-panel-content-header-day date-picker-day-title"
        >
          一
        </span>
        <span
          class="date-picker-panel-content-header-day date-picker-day-title"
        >
          二
        </span>
        <span
          class="date-picker-panel-content-header-day date-picker-day-title"
        >
          三
        </span>
        <span
          class="date-picker-panel-content-header-day date-picker-day-title"
        >
          四
        </span>
        <span
          class="date-picker-panel-content-header-day date-picker-day-title"
        >
          五
        </span>
        <span
          class="date-picker-panel-content-header-day date-picker-day-title"
        >
          六
        </span>
        <span
          class="date-picker-panel-content-header-day date-picker-day-title"
        >
          日
        </span>
      </div>
      <div>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-prev-month"
        >
          25
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-prev-month"
        >
          26
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-prev-month"
        >
          27
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-prev-month"
        >
          28
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-prev-month"
        >
          29
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-prev-month"
        >
          30
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          1
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          2
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          3
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          4
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          5
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          6
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          7
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          8
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          9
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          10
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          11
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          12
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          13
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          14
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          15
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          16
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          17
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          18
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          19
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          20
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          21
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          22
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month date-picker-selected-date"
        >
          23
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          24
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          25
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          26
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          27
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          28
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          29
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          30
        </span>
        <span
          class="date-picker-panel-content-date date-picker-current-month"
        >
          31
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-next-month"
        >
          1
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-next-month"
        >
          2
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-next-month"
        >
          3
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-next-month"
        >
          4
        </span>
        <span
          class="date-picker-panel-content-date date-picker-panel-content-next-month"
        >
          5
        </span>
      </div>
    </div>
    <div
      class="date-picker-panel-footer"
    >
      <div
        class="date-picker-panel-footer-today"
      >
        今天
      </div>
      <div
        class="date-picker-panel-footer-clear"
      >
        清除
      </div>
    </div>
  </div>,
]
`;
